src/common/statistics/StatisticsManager.ts

```diff
@@ -0,0 +1,394 @@
+    const allCurrentlyPenalizedRelays = this.extractUnique(this.allStakeManagerEvents.stakePenalizedEvents)
+    const allCurrentlyUnlockedRelays = new Set(
+      [...allEverUnlockedRelays]
+        .filter(it => !allCurrentlyWithdrawnRelays.includes(it))
+        .filter(it => !allCurrentlyPenalizedRelays.includes(it))
+    )
+    const allCurrentlyStakedRelays = new Set(
+      [...allEverStakedRelays]
+        .filter(it => !allEverUnlockedRelays.includes(it))
+        .filter(it => !allCurrentlyPenalizedRelays.includes(it))
+    )
+    return [
+      ...Array.from(allCurrentlyStakedRelays).map(address => {
+        return {
+          address,
+          status: RelayServerStakeStatus.STAKE_LOCKED
+        }
+      }),
+      ...Array.from(allCurrentlyUnlockedRelays).map(address => {
+        return {
+          address,
+          status: RelayServerStakeStatus.STAKE_UNLOCKED
+        }
+      }),
+      ...Array.from(allCurrentlyWithdrawnRelays).map(address => {
+        return {
+          address,
+          status: RelayServerStakeStatus.STAKE_WITHDRAWN
+        }
+      }),
+      ...Array.from(allCurrentlyPenalizedRelays).map(address => {
+        return {
+          address,
+          status: RelayServerStakeStatus.STAKE_PENALIZED
+        }
+      })
+    ]
+  }
+
+  async getRelayServersInfo (): Promise<RelayServerInfo[]> {
+    const relayServersInfo: Array<Promise<RelayServerInfo>> = []
+    const relaysByStatus = this.getRelaysStakeStatus()
+    for (const relay of relaysByStatus) {
+      relayServersInfo.push(this.gatherRelayInfo(relay.address, relay.status))
+    }
+    return await Promise.all(relayServersInfo)
+  }
+
+  async gatherRelayInfo (managerAddress: Address, stakeStatus: RelayServerStakeStatus): Promise<RelayServerInfo> {
+    let registrationInfo: RelayServerRegistrationInfo | undefined
+    const stakeManagerEvents = await this.getStakeManagerEvents(managerAddress)
+    const authorizedHubs = await this.getAuthorizedHubsAndVersions(stakeManagerEvents)
+
+    const relayRegisteredEvents =
+      this.allRelayHubEvents.relayRegisteredEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const transactionRelayedEvents =
+      this.allRelayHubEvents.transactionRelayedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+    const transactionRejectedEvents =
+      this.allRelayHubEvents.transactionRejectedEvents.filter(it => it.returnValues.relayManager === managerAddress)
+
+    const relayHubEvents: RelayHubEvents = {
+      relayRegisteredEvents,
+      transactionRelayedEvents,
+      transactionRejectedEvents
+    }
+
+    const isRegistered = stakeStatus === RelayServerStakeStatus.STAKE_LOCKED && relayRegisteredEvents.length !== 0
+    const relayHubEarningsBalance = (await this.contractInteractor.hubBalanceOf(managerAddress)).toString()
+    const stakeInfo = await this.contractInteractor.stakeManagerStakeInfo(managerAddress)
```

Comment:
AFAIK, you collected all info from all events, ever. now you do direct calls to collect the info again from the hub..